```
w1thermsensor> _________________________ test_repr_protocol[sensors0] _________________________
w1thermsensor> 
w1thermsensor> sensors = [{'id': '1', 'temperature': 20, 'type': <Sensor.DS18B20: 40>}]
w1thermsensor> 
w1thermsensor>     @pytest.mark.parametrize(
w1thermsensor>         "sensors",
w1thermsensor>         [
w1thermsensor>             # just a sensor
w1thermsensor>             ({"type": Sensor.DS18B20, "id": "1"},)
w1thermsensor>         ],
w1thermsensor>         indirect=["sensors"],
w1thermsensor>     )
w1thermsensor>     def test_repr_protocol(sensors):
w1thermsensor>         """Test the __repr__ protocol of a sensor instance"""
w1thermsensor>         # given
w1thermsensor>         sensor = W1ThermSensor()
w1thermsensor>         # when
w1thermsensor> >       sensor_copy = eval(repr(sensor))
w1thermsensor> 
w1thermsensor> tests/test_core.py:517: 
w1thermsensor> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
w1thermsensor> <string>:1: in <module>
w1thermsensor>     ???
w1thermsensor> /nix/store/44vp0f7j77ga3mkxwq3h3wvpd91qkn5i-python3.11-w1thermsensor-2.0.0/lib/python3.11/site-packages/w1thermsensor/core.py:150: in __init__
w1thermsensor>     self.BASE_DIRECTORY / (self.slave_prefix + self.id) / self.SLAVE_FILE
w1thermsensor> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
w1thermsensor> 
w1thermsensor> self = W1ThermSensor(sensor_type=40, sensor_id='1')
w1thermsensor> 
w1thermsensor>     @property
w1thermsensor>     def slave_prefix(self) -> str:
w1thermsensor>         """Returns the slave prefix for this temperature sensor"""
w1thermsensor> >       return "%s-" % hex(self.type.value)[2:]
w1thermsensor> E       AttributeError: 'int' object has no attribute 'value'
w1thermsensor> 
w1thermsensor> /nix/store/44vp0f7j77ga3mkxwq3h3wvpd91qkn5i-python3.11-w1thermsensor-2.0.0/lib/python3.11/site-packages/w1thermsensor/core.py:222: AttributeError
w1thermsensor> =========================== short test summary info ============================
w1thermsensor> FAILED tests/test_core.py::test_repr_protocol[sensors0] - AttributeError: 'int' object has no attribute 'value'
w1thermsensor> ======================== 1 failed, 144 passed in 1.05s =========================
```

Tests fail on python311. Best to report this upstream, if noone hasn't. We can disable the tests if it doesn't affect functionality, or disable the package on python311, if it does.

--- a/pkgs/development/python-modules/w1thermsensor/default.nix
+++ b/pkgs/development/python-modules/w1thermsensor/default.nix
@@ -1,0 +1,70 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pkgs
+, setuptools
+, aiofiles
+, click
+, coverage
+, tomli
+, pytest
+, pytest-mock
+, pytest-asyncio
+, pytestCheckHook
+, pythonOlder
+}:
+buildPythonPackage rec {
+  pname = "w1thermsensor";
+  version = "2.0.0";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-EcaEr4B8icbwZu2Ty3z8AAgglf74iZ5BLpLnSOZC2cE=";
+  };
+
+  postPatch = ''
+    sed -i 's/3\.5\.\*/3.5/' setup.py
+  '';
+
+  nativeBuildInputs = [
+    setuptools
+  ];
+
+  propagatedBuildInputs = [
+    aiofiles
+    click
+  ];
+
+  # Don't try to load the kernel module in tests.
+  env.W1THERMSENSOR_NO_KERNEL_MODULE = 1;
+
+  nativeCheckInputs = [
+    pytest-mock
+    pytest-asyncio
+    pytestCheckHook
+  ] ++ lib.optionals (pythonOlder "3.11") [
+    tomli
+  ];
+
+  # Tests for 2.0.0 currently fail on python3.11
+  # https://github.com/timofurrer/w1thermsensor/issues/116
+  doCheck = pythonOlder "3.11";
+
+  pythonImportsCheck = [
+    "w1thermsensor"
+  ];
+
+  meta = with lib; {
+    description = "Python interface to 1-Wire temperature sensors";
+    longDescription = ''
+      A Python package and CLI tool to work with w1 temperature sensors like
+      DS1822, DS18S20 & DS18B20 on the Raspberry Pi, Beagle Bone and other
+      devices.
+    '';
+    homepage = "https://github.com/timofurrer/w1thermsensor";
+    license = licenses.mit;
+    maintainers = with maintainers; [ quentin ];
+    platforms = platforms.all;
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -12459,6 +12459,8 @@ self: super: with self; {
 
   vyper = callPackage ../development/compilers/vyper { };
 
+  w1thermsensor = callPackage ../development/python-modules/w1thermsensor { };
+
   w3lib = callPackage ../development/python-modules/w3lib { };
 
   wadllib = callPackage ../development/python-modules/wadllib { };

